### PR TITLE
TOURCMS-9968 Update API JS Wrapper

### PR DESCRIPTION
**Pull request by armando-palisis**

### CODE REVIEW

File: tourcms.js [hunk 1]
```diff
@@ -1,799 +1,841 @@
...
+        verb = verb ?? TourCMS.HTTP_VERB_GET
+
+        let endpoint = this.baseURL + path
+
+        let outboundTime = this.time()
+        console.info(endpoint)
+        console.info(channelId)
+        console.info(verb)
+        console.info(outboundTime)
+        console.info(this.APIKey)
```
Comment (palisis-juanra) [inline tourcms.js:182]: Outputs are not needed

File: tourcms.js [hunk 2]
```diff
@@ -1,799 +1,841 @@
...
+    }
+
+    request(path, channelId, verb, postData = null) {
+
+        channelId = channelId ?? 0
+        verb = verb ?? TourCMS.HTTP_VERB_GET
+
+        let endpoint = this.baseURL + path
+
+        let outboundTime = this.time()
```
Comment (palisis-juanra) [inline tourcms.js:177]: Output is not needed

File: tourcms.js [hunk 3]
```diff
@@ -1,799 +1,841 @@
...
+        return (this.request(TourCMS.C_BOOKING_PAYMENT_SPREEDLY_NEW, channel, TourCMS.HTTP_VERB_POST, paymentData));
+    }
+
+    spreedlyCompletePayment(transactionId, channel) {
+        let endpoint = TourCMS.C_BOOKING_PAYMENT_SPREEDLY_COMPLETE + '?id=' + transactionId
+        return (this.request(endpoint, channel, TourCMS.HTTP_VERB_POST));
+    }
+
+    cancelBooking(channel, bookingId, note) {
+        let xmlString = "<booking><booking_id>" + bookingId + "</booking_id><note>" + note + "</note></booking>";
```
Comment (palisis-juanra) [inline tourcms.js:500]: Better to leave the user to generate the XML or being sure that we have all possible required fields

File: tourcms.js [hunk 4]
```diff
@@ -1,799 +1,841 @@
...
+    bookingUpdateComponent(componentData, channel) {
+        return (this.request(TourCMS.C_BOOKING_COMPONENT_UPDATE, channel, TourCMS.HTTP_VERB_POST, componentData));
+    }
+
+    bookingRemoveComponent(componentData, channel) {
+        return (this.request(TourCMS.C_BOOKING_COMPONENT_DELETE, channel, TourCMS.HTTP_VERB_POST, componentData));
+    }
+
+    addNoteToBooking(booking, channel, text, type) {
+        let xmlString = "<booking><booking_id>"+booking+"</booking_id><note><text>"+text+"</text><type>"+type+"</type></note></booking>";
```
Comment (palisis-juanra) [inline tourcms.js:528]: Better to leave the user to generate the XML  or being sure that we have all possible required fields

File: tourcms.js [hunk 5]
```diff
@@ -1,799 +1,841 @@
...
+        for (let field of fields) {
+
+            let xpathResult = xml.xpath(field);
+
+            for (let result of xpathResult) {
+                values.push(result[0].toString())
+            }
+        }
+
+        let stringToHash = values.join("|");
```
Comment (palisis-juanra) [inline tourcms.js:830]: ideally would be a default value and a param to override it, it would allow us more flexibility

**Review by palisis-juanra — CHANGES_REQUESTED** [review]

File: tourcms.js (hunk 4 above)
Comment (armando-palisis) [reply tourcms.js:528]: done

File: tourcms.js (hunk 3 above)
Comment (armando-palisis) [reply tourcms.js:500]: done, let user pass XML

File: tourcms.js (hunk 2 above)
Comment (armando-palisis) [reply tourcms.js:177]: done

File: tourcms.js (hunk 1 above)
Comment (armando-palisis) [reply tourcms.js:182]: done

File: tourcms.js (hunk 5 above)
Comment (armando-palisis) [reply tourcms.js:830]: this function is intended to be used with xml object with signature algorithm, and fields

**Review by palisis-juanra — CHANGES_REQUESTED** [review]
Looks good.
Please, can you confirm the XML2JSON function works fine.
If need to be changed by an external one, would be nice too or create a new one ensuring it works fine with XML arrays and specific delicated parts

**Review by palisis-juanra — CHANGES_REQUESTED** [review]
Please remove the node_modules folder.
Add the folder to gitignore
Please add a note indicating that request will return the object of the promise and the user must handle it to convernt

**Pull request merged**